include/libcyphal/application/registry/registry_impl.hpp

```diff
@@ -0,0 +1,215 @@
+    ///
+    /// @param name The name of the register. Should be unique within the registry.
+    /// @param options Extra options for the register, like "persistent" option (`true` by default).
+    /// @param getter The getter function to provide the register value (from a storage).
+    /// @param setter The setter function to update the register value (at the storage).
+    /// @return Result register if it was appended successfully. Otherwise, `nullopt`.
+    ///
+    template <typename Getter, typename Setter>
+    cetl::optional<Register<Getter, Setter>> route(const IRegister::Name     name,
+                                                   const IRegister::Options& options,
+                                                   Getter&&                  getter,
+                                                   Setter&&                  setter)
+    {
+        auto reg = makeRegister(memory(), name, options, std::forward<Getter>(getter), std::forward<Setter>(setter));
+        if (append(reg))
+        {
+            return reg;
+        }
+        return cetl::nullopt;
+    }
+
+    /// Constructs a read-write register, and links it to a given registry.
+    ///
+    /// A simple wrapper over route() that allows one to expose and mutate an arbitrary object as a mutable register.
+    ///
+    /// @param name The name of the register. Should be unique within the registry.
+    /// @param inout_value The referenced value; shall outlive the register.
+    /// @param options Extra options for the register, like "persistent" option.
+    /// @return Result register if it was appended successfully. Otherwise, `nullopt`.
+    ///
+    template <typename T>
+    auto expose(const IRegister::Name name, T& inout_value, const IRegister::Options& options = {})
+    {
+        return route(
+            name,
+            options,
+            [&inout_value]() -> const T& { return inout_value; },  // Getter
+            [&inout_value](const Value& v) {                       // Setter
+                //
+                inout_value = registry::get<T>(v).value();  // Guaranteed to be coercible by the protocol.
+                return true;
+            });
+    }
+
+    /// Constructs a parameter register, and links it to a given registry.
+    ///
+    /// In contrast to the above `expose()`, this method allows one to expose a parameter with a default value.
+    /// Exposed parameter value is stored inside the register and can be mutated (by default).
+    ///
+    /// @param name The name of the register. Should be unique within the registry.
+    /// @param default_value Initial default value.
+    /// @param options Extra options for the register, like "persistent" option.
+    /// @return Result register if it was appended successfully. Otherwise, `nullopt`.
+    ///
+    template <typename T, bool IsMutable = true>
+    cetl::optional<ParamRegister<T, IsMutable>> exposeParam(const IRegister::Name     name,
```

Comment:
Not sure I understand the purpose of this. Isn't this equivalent to:

```
T value = default;
auto register = registry.expose("name", value);
```

If this is meant to model the `P` class in Telega, then I think it shouldn't be here in this abstract registry module because it is too application-specific.

Reply:
Yes, it meant to model the `P`. I don't see it as application-specific. In contrast, I think it allows to expose a value as a register without worrying where the value is stored and its lifetime relative to the register - very light IMO, and without needed to store also 2 lambdas.

Reply:
Ok, then we need a different name for the method: not `exposeParam` but something like just `param[eter]`.

Reply:
>param
hmm... what about a verb in the method's name?
I skip verb mostly in very simple cases only - like returning just reference to a private member (see f.e. `Node::heartbeatProducer() -> node::HeartbeatProducer&`). What about `makeParam`?

Reply:
`makeParam` is okay but then maybe `parameterize`?

Reply:
ok, I'll go with "parameterize"
in the next PR